src/app/ipv/middleware.js

```diff
@@ -358,87 +341,54 @@ module.exports = {
+      end: "end",
+      "address-current": "address-current",
+      "attempt-recovery": "attempt-recovery",
+      "build-client-oauth-response": "build-client-oauth-response",
+      next: "next",
+      "next/end": "end",
+      "next/f2f": "f2f",
+      "next/dcmaw": "dcmaw",
+      "next/bank-account": "bankAccount",
+      "next/passport": "ukPassport",
+      "next/driving-licence": "drivingLicence",
+    };
     try {
-      checkForIpvAndOauthSessionId(req, res);
-
-      if (req.body?.journey === "end") {
-        await handleJourneyResponse(req, res, "end", currentPageId);
-      } else if (req.body?.journey === "addressCurrent") {
-        await handleJourneyResponse(req, res, "address-current", currentPageId);
-      } else if (req.body?.journey === "attempt-recovery") {
-        await handleJourneyResponse(
-          req,
-          res,
-          "attempt-recovery",
-          currentPageId,
-        );
-      } else if (req.body?.journey === "build-client-oauth-response") {
-        req.session.ipAddress = req?.session?.ipAddress
-          ? req.session.ipAddress
-          : getIpAddress(req);
-        await handleJourneyResponse(
-          req,
-          res,
-          "build-client-oauth-response",
-          currentPageId,
-        );
-      } else {
-        await handleJourneyResponse(req, res, "next", currentPageId);
-      }
-    } catch (error) {
-      transformError(error, `error handling POST request on ${currentPageId}`);
-      next(error);
-    }
-  },
-
-  handleMultipleDocCheck: async (req, res, next, currentPageId) => {
-    try {
-      checkForSessionId(req, res);
-
-      if (req.body?.journey === "next/passport") {
-        await handleJourneyResponse(req, res, "ukPassport", currentPageId);
-      } else if (req.body?.journey === "next/driving-licence") {
-        await handleJourneyResponse(req, res, "drivingLicence", currentPageId);
+      const action = journeyActions[req.body?.journey] || "next";
+      if (pagesUsingSessionId.includes(currentPageId)) {
+        checkForSessionId(req, res);
       } else {
-        await handleJourneyResponse(req, res, "end", currentPageId);
+        checkForIpvAndOauthSessionId(req, res);
       }
-    } catch (error) {
-      transformError(error, `error handling POST request on ${currentPageId}`);
-      next(error);
-    }
-  },
-  handleEscapeM2b: async (req, res, next, currentPageId) => {
-    try {
-      checkForSessionId(req, res);
 
-      if (req.body?.journey === "next") {
-        await handleJourneyResponse(req, res, "next", currentPageId);
-      } else if (req.body?.journey === "next/bank-account") {
-        await handleJourneyResponse(req, res, "bankAccount", currentPageId);
-      } else {
-        await handleJourneyResponse(req, res, "end", currentPageId);
+      if (action === "build-client-oauth-response") {
+        req.session.ipAddress = req?.session?.ipAddress
+          ? req.session.ipAddress
+          : getIpAddress(req);
```

Comment:
This seems suspicious, but I'm scared of changing anything in analytics without fully understanding it! (I'm a bit surprised we don't just set the IP address on every request independently...)

Leave this in for now :)